testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -105,6 +113,70 @@ def test_check_set_unit_nolength(self, D):
         with pytest.raises(ValueError):
             D._check_set_unit(units)
 
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
+        assert D._check_set_unit(units) is None
```

Comment:
Sorry I'm going to have to query this again, I don't see how asserting `None` on a function that returns nothing does anything else but flush the code path?

Is this just checking lines 755-756?

If so this is what I would do here:

```suggestion
        D._check_set_unit(units)
        assert D.units['density'] is None
```